runtime/src/status_cache.rs

```diff
@@ -131,13 +131,13 @@ impl<T: Clone> StatusCache<T> {
     pub fn get_signature_status_all<U>(
         checkpoints: &[U],
         signature: &Signature,
-    ) -> Option<Result<(), T>>
+    ) -> Option<(usize, Result<(), T>)>
     where
         U: Deref<Target = Self>,
     {
-        for c in checkpoints {
+        for (i, c) in checkpoints.iter().enumerate() {
```

Comment:
this basically says that a child bank implies a confirmation?

Reply:
Yea. I am going to store the stake weighted vector of parent confirmations for each bank fork.